plugins/cpu/ram-cpu/src/main/java/net/sf/emustudio/ram/cpu/impl/EmulatorEngine.java

```diff
@@ -214,7 +215,7 @@ private String arithmetic(String sym0, String sym1, ArithOp op) throws IOExcepti
                 try {
                     int rr0 = Integer.decode(r0);
                     if (rr0 == 0) {
-                        IP = rr0;
+                        IP = (Integer) in.getOperand();
```

Comment:
When doing a jump, the IP should be set to new address, NOT to the current value of R0 (accumulator).